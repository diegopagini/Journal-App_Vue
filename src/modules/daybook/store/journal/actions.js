/** @format */
import journalApi from '@/api/journalApi';

export const loadEntries = async (/*{ commit }*/) => {
	const { data } = await journalApi.get('/entries.json');
	console.log(data);
};

export const updateEntry = async (/* { commit }*/) => {};

export const createEntry = async (/* { commit }*/) => {};
