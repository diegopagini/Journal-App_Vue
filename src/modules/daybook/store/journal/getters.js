/** @format */

export const getEntriesByTerm =
	(state) =>
	(term = '') => {
		if (term.length === 0) return state.entries;

		return state.entries.filter((entry) => entry.text.toLowerCase().includes(term.toLowerCase()));
	};

export const getEntryById = (/* state */) => {};
