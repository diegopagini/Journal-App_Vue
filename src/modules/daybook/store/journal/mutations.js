/** @format */

export const setEntries = (state, entries) => {
	state.entries = [...state.entries, ...entries];
	state.isLoading = false;
};

export const updateEntry = (/*state*/) => {};

export const createEntry = (/*state*/) => {};
