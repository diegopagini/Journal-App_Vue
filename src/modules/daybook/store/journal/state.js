/** @format */

export default () => ({
	isLoading: false,
	entries: [
		{
			id: '1',
			date: new Date().toDateString(),
			text: 'Hola mundo',
			picture: null,
		},
		{
			id: '2',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '3',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '4',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
		{
			id: '5',
			date: new Date().toDateString(),
			text: 'Cillum sunt minim deserunt eu cupidatat deserunt fugiat culpa tempor sunt laborum reprehenderit aliqua sint. Do in ea reprehenderit do. Excepteur anim aliqua sunt deserunt et excepteur voluptate cupidatat labore labore excepteur. Minim commodo deserunt irure sunt consectetur irure fugiat tempor labore culpa ut sunt et. Enim non in velit quis culpa deserunt esse nostrud incididunt et qui consequat. Reprehenderit esse cillum eiusmod sint consequat non eu.',
			picture: null,
		},
	],
});
