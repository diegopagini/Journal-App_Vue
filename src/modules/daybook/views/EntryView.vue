<!-- @format -->

<template>
	<template v-if="entry">
		<div class="entry-title d-flex justify-content-between p-2">
			<div>
				<span class="text-succes fs-3 fw-bold">{{ day }}</span>
				<span class="mx-1 fs-3">{{ month }}</span>
				<span class="mx-2 fs-4 fw-light">{{ yearDay }}</span>
			</div>

			<div>
				<button class="btn btn-danger mx-2">Borrar <i class="fa fa-trash-alt"></i></button>
				<button class="btn btn-primary">Subir foto <i class="fa fa-upload"></i></button>
			</div>
		</div>

		<hr />

		<div class="d-flex flex-column px-2 h-75">
			<textarea
				placeholder="¿Qué sucedió hoy?"
				v-model="entry.text"
			></textarea>
		</div>

		<Fab
			icon="fa-save"
			@on:click="saveEntry"
		/>

		<img
			src="https://images.pexels.com/photos/1287145/pexels-photo-1287145.jpeg?cs=srgb&dl=pexels-eberhard-grossgasteiger-1287145.jpg&fm=jpg"
			alt="entry-picture"
			class="img-thumbnail"
		/>
	</template>
</template>
<script>
	import { defineAsyncComponent } from 'vue';
	import { mapGetters, mapActions } from 'vuex';
	import getDayMonthYear from '../helpers/getDayMonthYear';

	export default {
		props: {
			id: {
				type: String,
				required: true,
			},
		},
		components: {
			Fab: defineAsyncComponent(() => import('../components/fab-entry.vue')),
		},
		data() {
			return {
				entry: null,
			};
		},
		methods: {
			...mapActions('journal', ['updateEntry']),
			loadEntry() {
				const entry = this.getEntryById(this.id);
				if (!entry) return this.$router.push({ name: 'no-entry' });

				this.entry = entry;
			},
			async saveEntry() {
				this.updateEntry(this.entry);
			},
		},
		computed: {
			...mapGetters('journal', ['getEntryById']),
			day() {
				const { day } = getDayMonthYear(this.entry.date);
				return day;
			},
			month() {
				const { month } = getDayMonthYear(this.entry.date);
				return month;
			},
			yearDay() {
				const { yearDay } = getDayMonthYear(this.entry.date);
				return yearDay;
			},
		},
		created() {
			this.loadEntry();
		},
		watch: {
			id(value, oldValue) {
				if (value !== oldValue) this.loadEntry();
			},
		},
	};
</script>

<style lang="scss" scoped>
	textarea {
		border: none;
		font-size: 20px;
		height: 100%;

		&:focus {
			outline: none;
		}
	}

	img {
		bottom: 150px;
		box-shadow: 0px 5px 10px rgba($color: #000000, $alpha: 0.2);
		position: fixed;
		right: 20px;
		width: 200px;
	}
</style>
