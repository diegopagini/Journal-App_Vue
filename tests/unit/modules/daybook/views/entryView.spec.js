/** @format */
import journal from '@/modules/daybook/store/journal';
import EntryView from '@/modules/daybook/views/EntryView.vue';
import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';

import { journalState } from '../../../mock-data/test-journal-state';

const createVuexStore = (initialState) =>
	createStore({
		modules: {
			journal: {
				...journal,
				state: { ...initialState },
			},
		},
	});

describe('EntryView Component', () => {
	let wrapper;

	const mockRouter = {
		push: jest.fn(),
	};

	const store = createVuexStore(journalState);

	beforeEach(() => {
		jest.clearAllMocks();

		wrapper = shallowMount(EntryView, {
			props: {
				id: '-NbnHbDAWZvXdrl1ZZei',
			},
			global: {
				mocks: {
					$router: mockRouter,
				},
				plugins: [store],
			},
		});
	});

	it('Should kick out the user if the user not exist', () => {
		shallowMount(EntryView, {
			global: {
				mocks: {
					$router: mockRouter,
				},
				plugins: [store],
			},
			props: {
				id: 'This ID not exist',
			},
		});

		expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
	});

	it('Should show the entry correctly', () => {
		expect(wrapper.html()).toMatchSnapshot();
		expect(mockRouter.push).not.toHaveBeenCalled();
	});
});
