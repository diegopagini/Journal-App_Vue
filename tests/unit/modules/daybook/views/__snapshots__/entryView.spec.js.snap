// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryView Component Should show the entry correctly 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-succes fs-3 fw-bold">14</span><span class="mx-1 fs-3">Agosto</span><span class="mx-2 fs-4 fw-light">2023, Lunes</span></div><input type="file" accept="image/png, image/jpeg" style="display: none;">
  <div><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-2 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<fab-stub icon="fa-save"></fab-stub>
<img src="https://res.cloudinary.com/dfhtj8isz/image/upload/v1692006702/rlktggh4wrlllibwta2l.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
`;
